From 9e61c83b96691abb6307edf391ebdbfface02dfd Mon Sep 17 00:00:00 2001
From: Jonas Kvinge <jonas@jkvinge.net>
Date: Thu, 12 Oct 2023 19:17:20 +0200
Subject: [PATCH] macdeployqt: Fix codesigning with @loader_path

When the -executable parameter is specified, macdeployqt uses
@loader_path's instead of @rpath's, this case was not handled in
getBinaryDependencies() used for the codesigning.

Fixes: QTBUG-118075
Pick-to: 6.6
Change-Id: Ie1e0d0781305e1849df9ec0d5fb1c3ce6713a62b
---
 src/tools/macdeployqt/shared/shared.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/tools/macdeployqt/shared/shared.cpp b/src/tools/macdeployqt/shared/shared.cpp
index 413f8413f69f..b7ee93f6fc1a 100644
--- a/src/tools/macdeployqt/shared/shared.cpp
+++ b/src/tools/macdeployqt/shared/shared.cpp
@@ -598,6 +598,10 @@ QStringList getBinaryDependencies(const QString executablePath,
             QString binary = QDir::cleanPath(executablePath + trimmedLine.mid(QStringLiteral("@executable_path/").length()));
             if (binary != path)
                 binaries.append(binary);
+        } else if (trimmedLine.startsWith("@loader_path/")) {
+            QString binary = QDir::cleanPath(QFileInfo(path).path() + "/" + trimmedLine.mid(QStringLiteral("@loader_path/").length()));
+            if (binary != path)
+                binaries.append(binary);
         } else if (trimmedLine.startsWith("@rpath/")) {
             if (!rpathsLoaded) {
                 rpaths = getBinaryRPaths(path, true, executablePath);
-- 
2.16.3

